src/main.ts

```diff
@@ -545,6 +546,30 @@ export async function getToken(input: CspInput): Promise<string> {
   }
 }
 
+export async function checkTokenExpiration(input: CspInput): Promise<string> {
+  const response = await cspClient.post("https://console.cloud.vmware.com/csp/gateway/am/api/auth/api-tokens/details", {
+    headers: {
+      "Content-Type": "application/json",
+      tokenValue: "$CSP_API_TOKEN",
+    },
+  })
+
+  cachedCspToken = {
```

Comment:
I'm not sure you can use what you get back from that request as a token for talking to our APIs. Better not to cache it here.